andela/balder-ah-backend
# #162608538 View trending tags functionality

#### What does this PR do?
This PR handles retrieving all tags and trending or most used tags


#### Description of Task to be completed?
Have the following endpoint working
GET /api/tags
To retrieve all the tags on the platform
GET /api/tags/trending
To retrieve all the trending tags based on most used tags


#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a POST request to **localhost:3000/api/articles** to post a new article with tags and add the required fields (title, description, body, tags(should be an array))
- Send a Get request to **localhost:3000/api/tags** to retrieve all the tags that have been added respectively using postman as a client
- Send Get request to **localhost:3000/api/tags/trending** to retrieve all the trending tags(tags that have been used the most).
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**


#### Any background context you want to provide?
A new article has to be posted with tags to retrieve all tags or trending tags.


#### What are the relevant pivotal tracker stories? (If applicable)
[#162608538](https://www.pivotaltracker.com/story/show/162608538)

#### Screenshots (If applicable)
<img width="1440" alt="screen shot 2018-12-17 at 9 19 24 am" src="https://user-images.githubusercontent.com/29441898/50074760-30d24e80-01dd-11e9-8da1-1e849d6e6ed5.png">
<img width="1440" alt="screen shot 2018-12-17 at 9 19 49 am" src="https://user-images.githubusercontent.com/29441898/50074761-316ae500-01dd-11e9-89f1-1441a18c28c4.png">



#### Questions:
